app/models/concerns/comment/mentioning.rb

```diff
@@ -13,7 +13,7 @@ module Mentioning
           team
         )
       ) |
-      ( @([-\w\d]+) )     # match mentioned users
+      ( @(#{User.ALLOWED_LOGIN_CHARACTERS}+) )     # match mentioned users
```

Comment:
@astopy access the constant via `User::ALLOWED_LOGIN_CHARACTERS` not with the dot notation for a method.